File: Data Structures & Algorithms/valid-binary-search-tree/submission-4.cpp
#include <limits>

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool helper(TreeNode* root, long long lowest, long long highest) {
        if(root == nullptr) {
            return true;
        }
        //check current node
        if(root->val >= highest || root->val <= lowest) {
            return false;
        }
        return(helper(root->left, lowest, root->val) && helper(root->right, root->val, highest));
    }

    bool isValidBST(TreeNode* root) {
        if(root == nullptr) {
            return true;
        }
        return helper(root, numeric_limits<long long>::min(), numeric_limits<long long>::max());
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        m = 1
        current = 1
        k = 1
        s = set(nums)
        visited = set()
        for val in s:
            if val in visited:
                continue
            visited.add(val)
            while True:
                if val + k not in s and val - k not in s:
                    break
                if val + k in s:
                    current += 1
                    visited.add(val + k)
                if val - k in s:
                    current += 1
                    visited.add(val - k)
                k += 1

            m = max(m, current)
            k = 1
            current = 1
        return m

            